### PR TITLE
Add Notion Query Database action

### PR DESCRIPTION
**Pull request by vunguyenhung**
This PR adds Notion query a database action to support this request: https://github.com/PipedreamHQ/pipedream/issues/2570

Notion API Doc can be found here: https://developers.notion.com/reference/post-database-query

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by dylburger** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

File: components/notion/actions/query-database/query-database.mjs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      optional: true
+    },
+    sorts: {
+      type: "object",
+      description: "Sort Pages in the Database [See the docs](https://developers.notion.com/reference/post-database-query-sort)",
+      optional: true
+    }
+  },
+  async run({ $ }) {
+    const params = { filter, sorts }
```
Comment (vellames-turing) [inline components/notion/actions/query-database/query-database.mjs:29]: You need to use `this` in order to get the `props` params
```suggestion
    const params = {
    filter: this.filter,
    sorts: this.sorts
    }
```

File: components/notion/actions/query-database/query-database.mjs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    notion,
+    databaseId: {
+      propDefinition: [
+        notion,
+        "databaseId",
+      ],
+    },
+    filter: {
+      type: "object",
+      description: "Filter Pages in the Database [See the docs](https://developers.notion.com/reference/post-database-query-filter)",
```
Comment (vellames-turing) [inline components/notion/actions/query-database/query-database.mjs:19]:
```suggestion
      description: "A filter is a single condition used to specify and limit the entries returned from a database query. Database queries can be filtered by page property values [See the docs](https://developers.notion.com/reference/post-database-query-filter)",
```

File: components/notion/actions/query-database/query-database.mjs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      optional: true
+    },
+    sorts: {
+      type: "object",
+      description: "Sort Pages in the Database [See the docs](https://developers.notion.com/reference/post-database-query-sort)",
```
Comment (vellames-turing) [inline components/notion/actions/query-database/query-database.mjs:24]:
```suggestion
      description: "A sort is a condition used to order the entries returned from a database query. [See the docs](https://developers.notion.com/reference/post-database-query-sort)",
```

**Review by vellames-turing — CHANGES_REQUESTED** [review]
Hi there, thanks for your contribution. I left some comments

**Comment by vunguyenhung** [comment]
Thanks @vellames-turing, I've address your comments! Kindly check again when you have time 🙏

**Comment by andrewjschuang** [comment]
@vunguyenhung @dannyroosevelt I think this was resolved in #2718

**Comment by dannyroosevelt** [comment]
> @vunguyenhung @dannyroosevelt It seems that #2718 resolved this a few days ago

Oh nice, so is this PR not needed?

**Comment by andrewjschuang** [comment]
> > @vunguyenhung @dannyroosevelt It seems that #2718 resolved this a few days ago
> 
> Oh nice, so is this PR not needed?

Yeah, it can be closed

**Pull request closed without merging**